### PR TITLE
Add support for Signal messages

### PR DESCRIPTION
**Pull request by gutjuri**
This PR adds support for mining signal message logs. I managed to extract Signal chat logs using this (https://github.com/carderne/signal-export) tool. The extracted chat logs can then be parsed using the novel `SignalParser` class.

If you like this PR, please consider adding the `hacktoberfest-accepted` tag :) Thanks.

### CODE REVIEW

File: chatminer/chatparsers.py [hunk 1]
```diff
@@ -140,3 +187,4 @@ class StartOfDateType(Enum):
 
 
 DATEREG = r'^((\d{1})|(\d{2}))((\.)|(\/))((\d{1})|(\d{2}))((\.)|(\/))((\d{2}))'
+DATEREG_SIGNAL = r'^\[\d{4}-\d{2}-\d{2} \d{2}:\d{2}\]'
```
Comment (joweich) [inline chatminer/chatparsers.py:190]: @gutjuri please refactor these variables into class variables of the respective parser classes

File: chatminer/chatparsers.py [hunk 2]
```diff
@@ -6,8 +6,7 @@
 from pathlib import Path
 logging.basicConfig(level=logging.INFO)
 
-
-class WhatsAppParser():
+class Parser():
```
Comment (joweich) [inline chatminer/chatparsers.py:9]: The methods `_parse_message`, `_read_file_into_list`, and `parse_file_into_df` are present in both derived classes. The code would be even cleaner if you introduced these methods as abstract methods (see [here](https://www.geeksforgeeks.org/abstract-classes-in-python/)) in the parent class to make the logic more rigid.

File: chatminer/chatparsers.py (hunk 1 above)
Comment (joweich) [inline chatminer/chatparsers.py:190]: please refactor these variables into class variables of the respective parser classes

**Review by joweich — CHANGES_REQUESTED** [review]
Great work @gutjuri! Left two suggestions to make the code even cleaner.

File: chatminer/chatparsers.py (hunk 2 above)
Comment (gutjuri) [reply chatminer/chatparsers.py:9]: true, but `_parse_message` and `_read_file_into_list` are not part of `Parser`'s public interface. In theory, there could be parsers (e.g. for telegram messages) that do not require these methods. I'd therefore advocate for only making `parse_file_into_df` an abstract method of `Parser`. However, it is your project, so if you really want me to, I can also include `_parse_message` and `_read_file_into_list` as abstract methods.

**Pull request merged**